### PR TITLE
Added travis ci support to ensure coverage

### PR DESCRIPTION
**Pull request by amilajack**

### CODE REVIEW

**Comment by amilajack** [comment]
@adambom also remember to enable the tests on [travis-ci.org](http://travis-ci.org/).

**Comment by amilajack** [comment]
@Sebmaster willing to offer significant contributions to this project. Is it still alive?

**Comment by Sebmaster** [comment]
Not really. I don't have access to the npm package, so can't put the improvements that are already in master there.

**Comment by amilajack** [comment]
I see. Maybe @adambom could give give you permission to publish.

**Comment by amilajack** [comment]
@Sebmaster I am willing to offer long term and active support for this project if that is fine with you.

I'm considering:

- Refactoring
- Offering test coverage
- Providing support for modern node versions
- Improving performance

Also may I have repo read/write permissions? That would make doing this a lot easier. Thanks!

**Comment by Sebmaster** [comment]
Sadly I have neither access to the npm package, nor the repo settings so I can't give you any of these  :disappointed:

**Comment by roobie** [comment]
Seems to me it is time to fork this repo, don't you agree?

**Comment by MaXwellFalstein** [comment]
Agreed! What is going on?

**Comment by Sebmaster** [comment]
Feel free, but since I'm neither actively using, contributing, nor maintaining parallel.js anymore, I don't think me forking it would be a good idea.

**Comment by MaXwellFalstein** [comment]
I have forked this repo a few times to different ends :)

I would like for it to be maintained by someone who is interested in maintaining it.

Is there any way for it to be opened to interested people?

**Comment by amilajack** [comment]
I was thinking about forking it but I dont think that's necessary. As long as @Sebmaster has read/write permissions, we can contribute and he can merge our PR's. And as for the npm package, we can publish a new npm package. Maybe `parallel.js-2`?

**Comment by MaXwellFalstein** [comment]
@amilajack I think an new generation two of parallel.js would be a good idea. We can mention the start point, but have everything we need. I am happy to host it and have you guys contribute as much as you like. Would anyone be up for that? If it is great, I will migrate it to my corporate GitHub account.

**Comment by MaXwellFalstein** [comment]
# Parallel2.js?

**Comment by amilajack** [comment]
Personally, I think @Sebmaster should own the npm package because he's been contributing the longest. @Sebmaster what do you think? @MaXwellFalstein but if you are considering being active with this project for the entirety of its lifetime, I don't see a problem with you owning the package.

**Comment by amilajack** [comment]
Actually if you could host the npm package and allow @Sebmaster, I, and others to have permissions to the package, that would be great.

**Comment by amilajack** [comment]
Accidentally closed this. Sorry about that

**Comment by Sebmaster** [comment]
I'd rather not own it. I don't have the maintenance bandwith to support it.

**Comment by amilajack** [comment]
@MaXwellFalstein alright I think you should own the package.

**Comment by amilajack** [comment]
@Sebmaster also what do you think about this PR? Is there anything I should do for it to get merged?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - node_modules
+
+install:
+  - npm install
+
+script:
+  - npm run test
+
+# Necessary to compile native modules for io.js v3 or Node.js v4
+env:
```
Comment (Sebmaster) [inline .travis.yml]: We don't use any native modules - why is this required?

**Comment by amilajack** [comment]
Removed it. Sorry about that.

**Comment by MaXwellFalstein** [comment]
Okay guys. I will check it out this evening and weekend. I will let you know what goes on.

I can put code towards it, but I am a full time student, it might be a bit hit and miss during certain times in the year.

I am happy to host and maintain the repo and NPM package. I can make it all accessible to you both, it will be the three of us in charge of pull requests. If you guys make edits, please do not make edits to the master branch.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+node_js:
+  - 6
+  - 4
+
+cache:
+  directories:
+    - node_modules
+
+install:
```
Comment (Sebmaster) [inline .travis.yml]: Doesn't travis do this automatically if you don't specify an install at all?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+sudo: false
+
+language: node_js
+
+node_js:
+  - 6
+  - 4
+
+cache:
```
Comment (Sebmaster) [inline .travis.yml]: I don't think our 2 small dependencies warrant caching / improve testing times enough.

**Comment by amilajack** [comment]
Removed caching. And I think by default, travis will test against the most stable version of node. But most of the projects I have seen usually have 4, 5, 6. I dont think supporting 5 is worth it because it isnt LTS

**Comment by Sebmaster** [comment]
Welp, just noticed I can't turn on travis for this repo. I think that's why I didn't merge in the beginning :disappointed:

**Comment by amilajack** [comment]
Hmm how about we add the travis file to the repo but run the tests in a fork of the repo. Obviously we dont get a lot of the benefits that way but at least we have some way to run tests.

**Comment by amilajack** [comment]
We should also run builds with appveyor. Also I dont think you need to be an admin to add a repo to appveyor.

**Comment by MaXwellFalstein** [comment]
We can do anything we want in Parallel2.js.

**Comment by MaXwellFalstein** [comment]
# [Parallel2.js is now live](https://github.com/MaXwellFalstein/Parallel2.js)

**Comment by amilajack** [comment]
Oh i thought we would keep this repo but create a new npm module. @Sebmaster doesnt have read/write to the npm module but he does to the repo. This would be much better tho. Should we redirect this repo to the new page?

**Comment by MaXwellFalstein** [comment]
I think we should do both of those and update the Wiki on this repo to tell users to look at the new repo.

**Comment by amilajack** [comment]
I just remembered that @Sebmaster doesnt have permissions to change the redirect. Also, I would recommend that when telling users to open the same issue in the new repo. That would make things easier.

**Comment by MaXwellFalstein** [comment]
I will link the new repo everything I ask someone to do the pull request on the new repo.

**Comment by amilajack** [comment]
Great! Even better.

**Comment by mathiasrw** [comment]
wow - I only see this thread now. 

@MaXwellFalstein and @Sebmaster - I have added you as members to the newly created parallel-js organisation. 

@MaXwellFalstein lets work together to get all your commits in https://github.com/MaXwellFalstein/Parallel2.js into this repo.

**Comment by mathiasrw** [comment]
Closed as #125 did the same
